sno/init.py

```diff
@@ -271,52 +453,83 @@ def iter_features_sorted(self, pk_callback, limit=None):
         # Print the Query Plan
         # dbcur.execute(f"""
         #     EXPLAIN QUERY PLAN
-        #     SELECT {gpkg.ident(self.table)}.*
+        #     SELECT {self.quote_ident(self.table)}.*
         #     FROM {tbl_name}
-        #         INNER JOIN {gpkg.ident(self.table)} ON ({tbl_name}.link={gpkg.ident(self.table)}.{gpkg.ident(self.primary_key)})
+        #         INNER JOIN {self.quote_ident(self.table)}
+        #         ON ({tbl_name}.link={self.quote_ident(self.table)}.{self.quote_ident(self.primary_key)})
         #     ORDER BY {tbl_name}.sort;
         # """)
         # print("\n".join("\t".join(str(f) for f in r) for r in dbcur.fetchall()))
 
         dbcur.execute(
             f"""
-            SELECT {gpkg.ident(self.table)}.*
+            SELECT {self.quote_ident(self.table)}.*
             FROM {tbl_name}
-                INNER JOIN {gpkg.ident(self.table)} ON ({tbl_name}.link={gpkg.ident(self.table)}.{gpkg.ident(self.primary_key)})
+                INNER JOIN {self.quote_ident(self.table)}
+                ON ({tbl_name}.link={self.quote_ident(self.table)}.{self.quote_ident(self.primary_key)})
             ORDER BY {tbl_name}.sort;
         """
         )
         click.echo(f"Ran SELECT query in {time.monotonic()-t2:0.1f}s")
         return dbcur
 
-    def iter_features(self):
-        dbcur = self.db.cursor()
-        dbcur.execute(f"SELECT * FROM {gpkg.ident(self.table)};")
-        return dbcur
-
     def build_meta_info(self, repo_version):
-        return gpkg.get_meta_info(self.db, layer=self.table, repo_version=repo_version)
+        db = gpkg.db(self.ogr_source)
+        return gpkg.get_meta_info(db, layer=self.table, repo_version=repo_version)
+
+
+def list_import_formats(ctx, param, value):
+    """
+    List the supported import formats
+    """
+    if not value or ctx.resilient_parsing:
+        return
+    names = set()
+    for prefix, ogr_driver_name in FORMAT_TO_OGR_MAP.items():
+        d = gdal.GetDriverByName(ogr_driver_name)
+        if d:
+            m = d.GetMetadata()
+            # only vector formats which can read things.
+            if m.get('DCAP_VECTOR') == 'YES' and m.get('DCAP_OPEN') == 'YES':
+                names.add(prefix)
+    for n in sorted(names):
+        click.echo(n)
+    ctx.exit()
 
 
 @click.command("import")
 @click.pass_context
-@click.argument("source", type=ImportPath())
+@click.argument("source")
 @click.argument(
     "directory",
     type=click.Path(file_okay=False, exists=False, resolve_path=True),
     required=False,
 )
+@click.option(
+    "--table",
+    "-t",
+    help="Which table to import. If not specified, this will be selected interactively",
```

Comment:
renaming of tables needs to come back

Reply:
it is still possible, and hasn't changed. 

`sno import --table=orig_table renamed_table/`

Reply:
It only needs to change if we need to remove the `--table` option and use unnamed arguments for table names. In which case it would have to be something like

```
sno import path/to.gpkg orig_table:renamed_table [orig_table2:renamed_table2]
```